be/src/olap/in_list_predicate.h

```diff
@@ -75,34 +79,281 @@ namespace doris {
+                _base_evaluate<false, false>(&column, nullptr, sel, size);
+            }
+        }
+    }
+
+    // todo(wb) support evaluate_and,evaluate_or
+    void evaluate_and(vectorized::IColumn& column, uint16_t* sel, uint16_t size,
+                      bool* flags) const override {
+        LOG(FATAL) << "IColumn not support in_list_predicate.evaluate_and now.";
+    }
+    void evaluate_or(vectorized::IColumn& column, uint16_t* sel, uint16_t size,
+                     bool* flags) const override {
+        LOG(FATAL) << "IColumn not support in_list_predicate.evaluate_or now.";
+    }
+
+private:
+    template <typename LeftT, typename RightT>
+    bool _operator(const LeftT& lhs, const RightT& rhs) const {
+        if constexpr (PT == PredicateType::IN_LIST) {
+            return lhs != rhs;
+        }
+        return lhs == rhs;
+    }
+
+    template <bool is_nullable>
+    void _base_evaluate(const ColumnBlock* block, uint16_t* sel, uint16_t* size) const {
+        uint16_t new_size = 0;
+        for (uint16_t i = 0; i < *size; ++i) {
+            uint16_t idx = sel[i];
+            sel[new_size] = idx;
+            const T* cell_value = reinterpret_cast<const T*>(block->cell(idx).cell_ptr());
+            if constexpr (is_nullable) {
+                new_size += _opposite ^ (!block->cell(idx).is_null() &&
+                                         _operator(_values.find(*cell_value), _values.end()));
+            } else {
+                new_size += _opposite ^ _operator(_values.find(*cell_value), _values.end());
+            }
+        }
+        *size = new_size;
+    }
+
+    template <bool is_nullable, bool is_and>
+    void _base_evaluate(const ColumnBlock* block, const uint16_t* sel, uint16_t size,
+                        bool* flags) const {
+        for (uint16_t i = 0; i < size; ++i) {
+            if (!flags[i]) {
+                continue;
+            }
+
+            uint16_t idx = sel[i];
+            const T* cell_value = reinterpret_cast<const T*>(block->cell(idx).cell_ptr());
+            auto result = true;
+            if constexpr (is_nullable) {
+                result &= !block->cell(idx).is_null();
+            }
+            result &= _operator(_values.find(*cell_value), _values.end());
+
+            if constexpr (is_and) {
+                flags[i] &= _opposite ^ result;
+            } else {
+                flags[i] |= _opposite ^ result;
+            }
+        }
+    }
+
+    template <bool is_nullable, bool is_opposite>
+    void _base_evaluate(const vectorized::IColumn* column,
+                        const vectorized::PaddedPODArray<vectorized::UInt8>* null_map,
+                        uint16_t* sel, uint16_t* size) const {
+        uint16_t new_size = 0;
+
+        auto decorator = [&](bool f) {
+            if constexpr (is_opposite) {
+                return !f;
+            }
+            return f;
+        };
+
+        auto inner_loop = [&](std::function<bool(int)> checker) {
```

Comment:
better not use lambda here. I have test ssb-flat 100g, the in predicate evaluate time increased from 1s500ms to 2s608ms